tools/commentMonitor/main.go

```diff
@@ -63,103 +101,137 @@ func main() {
-		args, err := regexValidation(*regex, *e.GetComment().Body)
-		if err != nil {
-			log.Printf("matching command not found")
-			os.Exit(78)
+			log.Println("author is a member or collaborator")
 		}
 
-		// Save args to file. Stores releaseVersion in ARG_0 and prnumber in ARG_1.
-		args = append(args, strconv.Itoa(prnumber))
-		writeArgs(args, *output)
+		// Extract args if regexString provided.
+		if cmClient.regexString != "" {
+			// To have unnamed arguments start at ARG_0 and so forth.
+			unnamedArgCount := 0
+
+			// Add comment arguments.
+			commentArgs := cmClient.regex.FindStringSubmatch(commentBody)[1:]
+			commentArgsNames := cmClient.regex.SubexpNames()[1:]
+			for i, argName := range commentArgsNames {
+				if argName == "" {
+					argName = fmt.Sprintf("ARG_%v", unnamedArgCount)
+					unnamedArgCount++
+				}
+				cmClient.allArgs[argName] = commentArgs[i]
+			}
 
-		envVars := make(map[string]string)
-		for _, e := range os.Environ() {
-			tmp := strings.Split(e, "=")
-			envVars[tmp[0]] = tmp[1]
-		}
+			// Add non-comment arguments if any.
+			cmClient.allArgs["PR_NUMBER"] = strconv.Itoa(prnumber)
 
-		var buf bytes.Buffer
-		commentTemplate := template.Must(template.New("Comment").Parse(os.Getenv(*templateEnvVar)))
-		if err := commentTemplate.Execute(&buf, envVars); err != nil {
-			log.Fatalln(err)
+			// write arguments to fs
+			err := cmClient.writeArgs()
+			if err != nil {
+				log.Fatalf("%v: could not write args to fs", err)
+			}
 		}
 
-		if err := postComment(clt, owner, repo, prnumber, buf.String()); err != nil {
-			log.Fatalln(err)
+		// Post generated comment to Github pr if COMMENT_TEMPLATE is set.
+		if os.Getenv("COMMENT_TEMPLATE") != "" {
+			// Add all env vars to allArgs.
+			for _, e := range os.Environ() {
+				tmp := strings.Split(e, "=")
+				cmClient.allArgs[tmp[0]] = tmp[1]
+			}
+			// Generate the comment template.
+			var buf bytes.Buffer
+			commentTemplate := template.Must(template.New("Comment").
+				Parse(os.Getenv("COMMENT_TEMPLATE")))
+			if err := commentTemplate.Execute(&buf, cmClient.allArgs); err != nil {
+				log.Fatalln(err)
+			}
+			// Post the comment.
+			if err := cmClient.ghClient.postComment(ctx, buf.String()); err != nil {
+				log.Fatalf("%v : couldn't post generated comment", err)
+			}
+			log.Println("comment successfully posted")
 		}
 
-		// Setting benchmark label.
-		benchmarkLabel := []string{"benchmark"}
-		if _, _, err := clt.Issues.AddLabelsToIssue(context.Background(), owner, repo, prnumber, benchmarkLabel); err != nil {
-			log.Fatalln(err)
+		// Set label to Github pr if LABEL_NAME is set.
+		if os.Getenv("LABEL_NAME") != "" {
+			if err := cmClient.ghClient.createLabel(ctx, os.Getenv("LABEL_NAME")); err != nil {
+				log.Fatalf("%v : couldn't set label", err)
+			}
+			log.Println("label successfully set")
 		}
 
 	default:
-		log.Fatalln("simpleargs only supports issue_comment event")
+		log.Fatalln("commentMonitor only supports issue_comment event")
```

Comment:
```suggestion
		log.Fatalln("only issue_comment event are supported")
```

lets not mention any names in case we rename the tool again.